csharp/src/IceLocatorDiscovery/PluginI.cs

```diff
@@ -206,48 +201,6 @@ public override Task<Ice.Object_Ice_invokeResult>
         }
     }
 
-    public List<Ice.LocatorPrx> getLocators(string instanceName, int waitTime)
-    {
-        //
-        // Clear locators from previous search.
-        //
-        lock (_mutex)
-        {
-            _locators.Clear();
-        }
-
-        //
-        // Find a locator
-        //
-        invoke(null, null);
-
-        //
-        // Wait for responses
-        //
-        if (instanceName.Length == 0)
-        {
-            Thread.Sleep(waitTime);
-        }
-        else
-        {
-            lock (_mutex)
-            {
-                while (!_locators.ContainsKey(instanceName) && _pending)
-                {
-                    Monitor.Wait(_mutex, waitTime);
```

Comment:
The only Monitor.Wait(_mutex, ...), which means all the Monitor.Pulse(_mutex) elsewhere are useless.